validator/utils.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Function that loops over an async function until it returns a truthy value.
+ * @param {Function} fn - Async function to itterate over
+ * @param {object} params - any params for the function.
+ * @param {string} inMsg - Log message to be displayed for each loop
+ * @param {string} outMsg - Log message to be displayed upon returning a value
+ * @returns {T} - returns generic value based fn return.
+ */
+export async function loop<T>(fn: Function, params: Array, inMsg: string, outMsg: string): T {
```

Comment:
Is this used anywhere?

Reply:
Ah shoot I meant to finish its accompanying test. Its suppose to be an interim solution to `polling` essentially replace the while loops everywhere.,.. I came up with it on the plane lol, thoughts?